### PR TITLE
change event and param syntax

### PR DESCRIPTION
**Pull request by UnsignedByte**
Implements #192.

### CODE REVIEW

**Comment by UnsignedByte** [comment]
https://github.com/cucapra/filament/issues/192#issuecomment-1698083759 mentions an issue with let-binding syntax that might need to be resolved.

**Comment by rachitnigam** [comment]
See https://github.com/cucapra/filament/issues/192#issuecomment-1698466139 for the answer. This change will require more work and we should probably not differentiate between the syntax of events and parameters at all.

**Review by gabizon103 — APPROVED** [review]
This looks good to me, but in regards to the discussion in #192, I think it might be worth having a clear (and not too ugly) syntactic distinction between params and events? I kind of like keeping `@` to denote events and leaving parameters with no prefix, instead of leaving them all with no prefix. I don't feel super strongly about this so if you want to remove all the prefixes then that would be cool too. This was just a thought.

**Comment by UnsignedByte** [comment]
Something like `@prod.G` might look better? I do think that having the `@` definitely helps as far as understanding a complex expression with parameters and events, as we can sort of disregard the event parts to an extent and only look at the offset.

**Comment by rachitnigam** [comment]
Hm, but then it's not very clear which thing is the event: `prod` or `G`. We could steal the Rust syntax for associated types:
```
m0 := M<G>(in);
m0::Ready // <- refers to the Ready event associated with the m0 invoke
```

However, we'd still need some syntax to access existentially defined parameters.

**Comment by rachitnigam** [comment]
Haven't reviewed yet but wanted to note that #304 might create conflicts.

File: src/ast/control.rs [hunk 1]
```diff
@@ -325,7 +325,7 @@ impl From<If> for Command {
 #[derive(Clone)]
 /// The type of the bundle:
 /// ```
-/// for<#i> @[G+#i, G+#i+1] #W
+/// for<#i> '[G+#i, G+#i+1] #W
```
Comment (rachitnigam) [inline src/ast/control.rs:328]: nit: the `'` should be inside the interval

File: src/ast/control.rs [hunk 2]
```diff
@@ -401,7 +401,7 @@ impl BundleType {
 #[derive(Clone)]
 /// Represents a bundle of wires with timing guarantees
 /// ```
-/// bundle f[10]: for<#i> @[G+#i, G+#i+1] #W;
+/// bundle f[10]: for<#i> '[G+#i, G+#i+1] #W;
```
Comment (rachitnigam) [inline src/ast/control.rs:404]: nit: same as above

**Review by rachitnigam — APPROVED** [review]
LGMT but we should make that the error messages generated in `ir::info` also use the new syntax. They manually construct strings for formatting parameters in some places so will look wrong.

**Pull request merged**